static/js/util.ts

```diff
@@ -206,28 +217,28 @@ export function sorted_ids(ids) {
     return id_list;
 }
 
-export function set_match_data(target, source) {
+export function set_match_data(target: Message, source: Message): void {
```

Comment:
The `source` argument isn’t necessarily a `Message`; it may [come](https://github.com/zulip/zulip/blob/747ea8e84d96e09aa5cf85b935ca66f54973e746/static/js/message_events.js#L42-L60) from the [`messages/matches_narrow` API](https://zulip.com/api/check-messages-match-narrow), which returns objects that only have `match_content` and `match_subject`.